### DISCUSSION TITLE
puppeteer not working with vercel

### DISCUSSION THREAD

**Discussion by karimaswan22** [vercel/vercel#4691] [Help] [answered]
> i have deployed and node.js with vecel and i used puppeteer and express but puppeteer not working 

> 
MY INDEX.JS :

`const puppeteer = require('puppeteer');
const express = require('express');
const server = express();

server.use(express.static("index.html"));
server.get('/' ,  async (req,res) =>{
		res.sendFile("index.html",{root:__dirname});

console.log("it running");
});
	

server.post('/' , async (req,res) =>{

console.log("button clicked server side");	
	(async () => {
const browser = await puppeteer.launch({
       headless: false,
		defaultViewport:null,
});
const page = await browser.newPage();
await page.goto('my-url-here');
let data = await page.evaluate(() => {
	name = document.querySelector("#name").innerText;
	 last = document.querySelector("#lastname").innerText;
return {
name,
last,
}
 });

data = ({
	"name":"aswan",
	"last":"abdelkarim",
	   });
res.send(data);
})();
});

function content() {
	
	console.log("listen to  3000 ");
}
server.listen(3000,content);
server.use(express.json());`

> MY HTML FILE
`<?xml version="1.0" encoding="utf-8"?>
<!DOCTYPE html>

<html>
<html lang="en">
<head>
<meta charset="utf-8"/>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8"/>

<title> hello </title>
</head>
<style>
</style>
<body>
<script src="index.js"></script>

<h1 id="namee"></h1>
<h1 id="lastnamee"></h1>
<button onclick="bnt()">start</button>
</body>
<script>
function bnt(){

let d = new XMLHttpRequest();
d.open("post","/",true);
d.onload = function (){
if(this.status == 200){
console.log("status = 200 ");
let data = JSON.parse(this.responseText);
console.log(data);
}else{
console.log("got error");

}

}

d.send();

}
</script>
</html>

`

> MY PACKAGE.JSON
{
  "name": "test puppeteer",
  "version": "1.0.0",
  "description": "",
  "main": "index.js",
  "dependencies": {
    "express": "^4.17.1",
    "fs": "0.0.1-security",
    "jest": "^22.4.0",
    "nodemon": "^2.0.4",
    "now": "^19.1.1",
    "puppeteer": "^4.0.0"
  },
  "devDependencies": {},
  "scripts": {
    "start": "now",
    "dev": "nodemon"
  },
  "keywords": [
    "aswan",
    "git",
    "js"
  ],
  "author": "hi",
  "license": "MIT",
  "optionalDependencies": {
    "bufferutil": "^4.0.1"
  }
}

> MY now.json
{

"version":2,
"regions":["sfo1"],
"builds":[
{
"src":"index.js",
"use":"@now/node-server"
}
],
"routes":[
{

"src": ".js",
"dest": "/index.js"

 

}
]
}

> puppeteer not working ...

**Comment by mcsdevv** [comment, marked as answer]
Hello,

The `@now/node-server` package is deprecated in favor of `@vercel/node`. We also do not recommend trying to run Puppeteer inside of a Serverless Function given its size and other issues that can arise when using it.

**Comment by karimaswan22** [comment]
i have changed @now/node-server with @now/node and it works but only if i run it in my  localhost that vervel provid but it's not working when i try to build a livehost https://good.blackaswan22.vercel.app/
it give me "Not Found" any other suggestions

**Comment by karimaswan22** [comment]
thank you for your reply sir i think that it works but if i try to build a
livehost with vercel i get in my website "Not Found"
[image: vvv.PNG]
